src/options.js

```diff
@@ -133,7 +133,17 @@ function switchNFLChanged(inputEvent) {
 }
 
 function switchWeatherChanged(inputEvent) {
-    localStorage.setItem(SHOW_WEATHER_OPTION.storageKey, inputEvent.target.checked)
+        localStorage.setItem(SHOW_WEATHER_OPTION.storageKey, inputEvent.target.checked)
+    if(inputEvent.target.checked === true){
+        document.getElementById('show-weather-api-key-tooltip').style.display = '';

```

Comment:
I believe this will change the value back to whatever it was before, but if you can provide any documentation on that I would really appreciate it.

Reply:
Yes, It will change the SHOW_WEATHER localStorage value to the opposite of what it is currently set to. Yes I can add some documentation. Do you have somewhere in mind on where to add it?

Reply:
Sorry, I think my comment was unclear. I know the `localStorage.setItem` stuff. My question was about the behavior of `style.display = ''`.

Suppose I start with `display=block`, then when I click this toggle it goes to `display=none`. When I toggle again, will it go back to `display=block` or something else?

Reply:
Ah sorry I understand now. yes it will go back to block the default is `block` so I used set it as ''. I have change it to 'block' for more clarity :)

Reply:
Would also appreciate if you let me know of any other ways you may of approached this?

Reply:
I think this is fine as is. I will start by saying I don't know what happens if you set it to empty versus block. If it is the case that javascript simply overrides the styles for the dom element, then setting to to empty is the best choice as you don't know or care what the other styles are. If it is the other case, we need to know what the appropriate value is (e.g. block vs online block). Now, this may very well be correct today, but a future change to those styles means this code needs to be updated, and that is likely to be overlooked.

One improvement code wise would be to have a function that sets each of those three dom elements' styles to a given value ('none' and 'block' in your two case). That way, it is clear that the intent is to manipulate that set of dom elements and it becomes easier to manage.

Overall though, this is fine.

Reply:
👍  Thanks for the feedback!